src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -2218,6 +2218,31 @@ public static bool IsIsOrAsContext(this SyntaxTree syntaxTree, int position, Syn
             // is/as are valid after expressions.
             if (token.IsLastTokenOfNode<ExpressionSyntax>())
             {
+                var expression = token.GetAncestor<ExpressionSyntax>();
+                var symbols = semanticModel.GetSymbolInfo(expression).GetBestOrAllSymbols();
+                var methodSymbols = symbols.OfType<IMethodSymbol>();
+                if (methodSymbols.Any())
+                {
+                    // Not after lambdas or anonymous methods
+                    if (methodSymbols.All(methodSymbol => methodSymbol.IsAnonymousFunction()))
+                    {
+                        return false;
+                    }
+                   
+                    // Not after expressions of type 'void'
+                    if (methodSymbols.All(methodSymbol => methodSymbol.ReturnsVoid && (methodSymbol.IsExtensionMethod() || methodSymbol.IsOrdinaryMethod())))
```

Comment:
❓ Is it possible for this to false-positive, e.g. in cases where the method name fails to resolve?